packages/cli/test/unit/util/input/project-selector.test.ts

```diff
@@ -0,0 +1,88 @@
+import { projectSelector } from '../../../../src/util/input/project-selector';
+import { MockClient } from '../../../mocks/client';
+import { setupFixtureFromTemplate } from '../../../helpers/setup-fixture';
+import path from 'path';
+import { RepoProjectsConfig } from '../../../../src/util/link/repo';
+
+describe('projectSelector', () => {
+  const rootFixturePath = path.join('commands', 'link', 'monorepo');
+  const expectedPaths = [
+    '/apps/app-1',
+    '/apps/app-2',
+    '/packages/package-1',
+    '/packages/package-2',
+  ];
+  const getExpectedPaths = (fixture: string) =>
+    fixture === 'nested'
+      ? expectedPaths.flatMap(p => [
+          path.join('/backend', p),
+          path.join('/frontend', p),
+        ])
+      : expectedPaths;
+
+  describe.each(['nested', 'npm', 'yarn', 'pnpm', 'nx', 'rush'])(
+    'fixture: %s',
+    fixture => {
+      let cwd: string;
+      let client: MockClient;
+      const expected = getExpectedPaths(fixture);
+
+      beforeEach(() => {
+        cwd = setupFixtureFromTemplate(
+          path.join(rootFixturePath, fixture),
+          path.join(rootFixturePath, '_template'),
+          fixture === 'nested' ? ['backend', 'frontend'] : undefined
+        );
+        client = new MockClient();
+      });
+
+      describe('when repo.json is not present', () => {
+        it('should return list of projects based on workspace list', async () => {
+          const projectSelectorPromise = projectSelector(client, cwd);
+          await expect(client.stderr).toOutput('Select projects');
+          client.stdin.write('a\n');
+          const actual = await projectSelectorPromise;
+          expect(actual.sort()).toStrictEqual(expected?.sort());
```

Comment:
**suggestion (non-blocking):**

```suggestion
          expect(actual.sort()).toStrictEqual(expected.sort());
```

It looks like `expected` will alway be an array of strings. I don't think we need to use `?.` here.